fitbenchmarking/results_processing/performance_profiler.py

```diff
@@ -171,27 +175,20 @@ def create_plot(ax, step_values, keys):
 
     no_failures = np.zeros(len(step_values), dtype=np.int8)
     huge = 1.0e20  # set a large value as a proxy for infinity
-    for i, solver_values in enumerate(step_values):
+    plot_points = np.linspace(0.0, 1.0, step_values[0].size)
+    plot_points = np.append(plot_points, 1.0)
+    
+    for i, (solver, solver_values) in enumerate(zip(labels, step_values)):
         inf_indices = np.where(solver_values > huge)
         solver_values[inf_indices] = huge
-        if inf_indices:
+        if inf_indices[0].size > 0:
             no_failures[i] = len(inf_indices[0])
-
-    labels = [key for key in keys]
-    for i, solver in enumerate(labels):
-        if no_failures[i]:
             labels[i] = "{} ({} failures)".format(solver, no_failures[i])
```

Comment:
```suggestion
            label = "{} ({} failures)".format(solver, len(inf_indices[0]))
```

Reply:
Fixed the `no_failures[i]`, but see comment below for the array `labels`

Reply:
fixed, as discussed